MigrationScript/AmsOperationsHelper.ps1

```diff
@@ -0,0 +1,294 @@
+# ########### Header ###########
+# Refer common library file
+. $PSScriptRoot\ConsoleLogger.ps1
+. $PSScriptRoot\UtilityFunctions.ps1
+. $PSScriptRoot\KeyvaultHelperFunctions.ps1
+# #############################
+
+<#
+.SYNOPSIS
+Function to create a new Provider in AMS v2 Monitor.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER request
+request object, it should have Name Property that has provider name and body property which has provider properties. 
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$requestObj = @{
+				Name = $providerName
+				body = @{
+					properties = @{
+						providerSettings = @{
+							providerType = $providerType
+							hostname = $Hostname
+							dbName = $DbName
+							sqlPort = $SqlPort
+							dbUsername = $DbUsername
+							dbPassword = $DbPassword
+						}
+					}
+				}
+			}
+$logger = New-Object ConsoleLogger
+PutAmsV2Provider -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -request $requestObj -logger $logger;
+#>
+function PutAmsV2Provider([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, $request, $logger) {
+	$rawToken = Get-AzAccessToken -ResourceTypeName Arm;
+    $armToken = $rawToken.Token;
+    $apiVersion = "2021-12-01-preview";
+
+	$headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $armToken"
+    };
+
+	$providerName = $request.name;
+    $bodyStr = $request.body | ConvertTo-Json -Depth 3;
```

Comment:
Should we first validate that teh request.Body has the requisite length etc ?

Reply:
We are creating the body in the function where it is called so some data is always going to be there. Not sure how to get json length. There might be some gaps in my understanding

Reply:
Resolving the conversation for now. Let me know if there is some way for standard indentation